allennlp/data/token_indexers/token_characters_indexer.py

```diff
@@ -41,6 +42,13 @@ def __init__(self,
                  start_tokens: List[str] = None,
                  end_tokens: List[str] = None,
                  min_padding_length: int = 0) -> None:
+        if min_padding_length == 0:
+            url = "https://github.com/allenai/allennlp/issues/1954"
+            warnings.warn("You are using the default value (0) of `min_padding_length`, "
+                          f"which can cause some subtle bugs (more info see {url}). "
+                          "Strongly recommend to set a value, usually the maximum size "
+                          "of the convolutional layer size when using CnnEncoder.",
+                          FutureWarning)
```

Comment:
I don't think `FutureWarning` is right here - nothing is going to change in the future, it's just a warning that the user might have buggy code.  Maybe the default of `UserWarning` is correct here?